### PR TITLE
Refactor of error handlings and add more async/await 'magic' to those…

### PR DESCRIPTION
**Pull request by ivanzusko**
In this **PR**:
- was refactored error handling: now the user will see brief info about what exactly went wrong;
- was added more async/await 'magic' to those parts where it was missed;

### CODE REVIEW

File: src/store/actions/api.actions.js [hunk 1]
```diff
@@ -179,15 +179,6 @@ export const fetchEndpoint = pathname => async dispatch => {
...
-        if (!response) {
-            dispatch(openResponseModal({
-                message: 'Something went wrong...',
-            }));
-
-            return;
-        }
-
         const preparedPlugins = response.data.plugins.map(plugin => {
             if (plugin.name === 'rate_limit') {
```
Comment (rafaeljesus) [inline src/store/actions/api.actions.js]: you could user switch case

File: src/store/actions/api.actions.js [hunk 2]
```diff
@@ -398,123 +392,62 @@ export const deleteEndpoint = apiName => dispatch => {
...
-                // eslint-disable-next-line
-                console.log(error.request);
-            } else {
-            // Something happened in setting up the request that triggered an Error
-                // eslint-disable-next-line
-                console.log('Error', error.message);
-            }
-        });
+    try {
+        const response = await client.put(`apis${pathname}`, preparedApi);
```
Comment (rafaeljesus) [inline src/store/actions/api.actions.js]: response is useless in this case right? Having `await client.put(`apis${pathname}`, preparedApi);` only would do the trick

File: src/store/actions/api.actions.js [hunk 3]
```diff
@@ -398,123 +392,62 @@ export const deleteEndpoint = apiName => dispatch => {
...
+        }));
+    }
 };
 
 export const confirmedDeleteEndpoint = async (dispatch, apiName) => {
     dispatch(deleteEndpointRequest());
+    dispatch(closeConfirmationModal());
 
     try {
         const response = await client.delete(`apis/${apiName}`);
```
Comment (rafaeljesus) [inline src/store/actions/api.actions.js]: same as [line 427](https://github.com/hellofresh/janus-dashboard/pull/134/files#diff-dc26439d13389fc50f2e25050dffee02R427)

File: src/store/api/index.js [hunk 4]
```diff
@@ -46,6 +46,8 @@ if (getAccessToken()) {
 client.interceptors.response.use(undefined, (error) => {
     if (error.response.status === 401) {
         history.push('/login');
+
+        return;
     }
     // if (error.response.status === 401 && error.config && !error.config.isRetryRequest) {
```
Comment (rafaeljesus) [inline src/store/api/index.js]: you remove this code if not used

File: src/store/api/index.js [hunk 5]
```diff
@@ -46,6 +46,8 @@ if (getAccessToken()) {
 client.interceptors.response.use(undefined, (error) => {
```
Comment (rafaeljesus) [inline src/store/api/index.js]: `error => {}`

File: src/store/actions/api.actions.js (hunk 2 above)
Comment (ivanzusko) [reply src/store/actions/api.actions.js]: `true`

File: src/store/actions/api.actions.js [hunk 6]
```diff
@@ -248,13 +239,14 @@ export const fetchEndpoint = pathname => async dispatch => {
 
             return plugin;
         });
-
         const lens = R.lensPath(['plugins']);
         const preparedApi = R.set(lens, preparedPlugins, response.data);
 
         dispatch(getEndpointSuccess(preparedApi, response.data));
```
Comment (rafaeljesus) [inline src/store/actions/api.actions.js]: I just wonder if in these cases we could compose 🤔 
```js
R.compose(
  dispatch,
  getEndpointSuccess
)(preparedApi, response.data)
```

File: src/store/actions/api.actions.js [hunk 7]
```diff
@@ -248,13 +239,14 @@ export const fetchEndpoint = pathname => async dispatch => {
...
     } catch (error) {
-        console.log('FETCH_ENDPOINT_ERROR', 'Infernal server error', error);
+        dispatch(openResponseModal({
```
Comment (rafaeljesus) [inline src/store/actions/api.actions.js]: Here too
```js
R.compose(
  dispatch,
  openResponseModal
)({ message: error.response.data.error })
```

File: src/store/actions/api.actions.js (hunk 6 above)
Comment (ivanzusko) [reply src/store/actions/api.actions.js]: @rafaeljesus , I'll check this out! 👍

File: src/store/api/index.js [hunk 8]
```diff
@@ -43,18 +43,16 @@ if (getAccessToken()) {
     setAccessToken(getAccessToken());
 }
 
-client.interceptors.response.use(undefined, (error) => {
+client.interceptors.response.use(undefined, error => {
     if (error.response.status === 401) {
+        console.warn('Here');
```
Comment (rafaeljesus) [inline src/store/api/index.js]: 😆

Comment (ivanzusko) [reply src/store/api/index.js]: Pora!!!

Comment (rafaeljesus) [reply src/store/api/index.js]: hahaha

**Comment by ivanzusko** [comment]
Ready, senior @rafaeljesus 😄 🇧🇷

**Pull request merged**